node/src/tasks/msp_move_bucket.rs

```diff
@@ -0,0 +1,312 @@
+        info!(
+            target: LOG_TARGET,
+            "MSP: user requested to move bucket {:?} to us",
+            event.bucket_id,
+        );
+
+        // Get the indexer database pool. If we don't have it, we can't continue.
+        let indexer_db_pool = if let Some(indexer_db_pool) =
+            self.storage_hub_handler.indexer_db_pool.clone()
+        {
+            indexer_db_pool
+        } else {
+            error!(
+                target: LOG_TARGET,
+                "Indexer is disabled but a move bucket event was received. Please provide a database URL (and enable indexer) for it to use this feature."
+            );
+
+            let call = storage_hub_runtime::RuntimeCall::FileSystem(
+                pallet_file_system::Call::msp_respond_move_bucket_request {
+                    bucket_id: event.bucket_id,
+                    response: BucketMoveRequestResponse::Rejected,
+                },
+            );
+
+            self.storage_hub_handler
+                .blockchain
+                .send_extrinsic(call, Tip::from(0))
+                .await?
+                .with_timeout(Duration::from_secs(
+                    self.storage_hub_handler
+                        .provider_config
+                        .extrinsic_retry_timeout,
+                ))
+                .watch_for_success(&self.storage_hub_handler.blockchain)
+                .await?;
+
+            return Ok(());
+        };
+
+        // TODO: check that we have enough space to accept the bucket and reject if not (+test)
+        // TODO: check that are know all the BSPs for the files in the bucket and reject if not (+test)
+
+        let call = storage_hub_runtime::RuntimeCall::FileSystem(
+            pallet_file_system::Call::msp_respond_move_bucket_request {
+                bucket_id: event.bucket_id,
+                response: BucketMoveRequestResponse::Accepted,
+            },
+        );
+
+        info!(
+            target: LOG_TARGET,
+            "MSP: accepting move bucket request for bucket {:?}",
+            event.bucket_id,
+        );
+
+        self.storage_hub_handler
+            .blockchain
+            .send_extrinsic(call, Tip::from(0))
+            .await?
+            .with_timeout(Duration::from_secs(
+                self.storage_hub_handler
+                    .provider_config
+                    .extrinsic_retry_timeout,
+            ))
+            .watch_for_success(&self.storage_hub_handler.blockchain)
+            .await?;
+
+        let bucket = event.bucket_id.as_ref().to_vec();
+
+        let mut indexer_connection = indexer_db_pool.get().await?;
+
+        let forest_storage = self
+            .storage_hub_handler
+            .forest_storage_handler
+            .get_or_create(&event.bucket_id.as_ref().to_vec())
+            .await;
+
+        for file in shc_indexer_db::models::File::get_by_onchain_bucket_id(
```

Comment:
Is this downloading one chunk per fail at a time? Shouldn't we optimise this so that it sends download requests to many BSPs for all the chunks of all the files at once?

Reply:
I know that could get cumbersome to handle errors then, but it would be infinitely faster.

Reply:
Anyway, if this works for now, it's fine, but we should consider this as an important improvement.

Reply:
Yes, that was the plan. Wrote it so that we can change it easily later, but didn't want to throw it in right now. Let's map this into a task as well.